### PR TITLE
Improved Climatic >Inventory dialog

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes Partially #8342 
@rdstern , @lloyddewit @N-thony , Fixes Part **a**, **b** and **c** of the issue. Part **d** will be fixed during the sprint with help of @lilyclements

### CODE REVIEW

**Comment by rdstern** [comment]
@Vitalis95 a) seems to be working fine.
b) Has now the correct settings in the dialog, - of month and Date order, but the code was still the same code as before - so not consistewnt with the new dialog.  See below that it is still the code by Day.

image

c) When I tried to save - to check the new key - I got the following error:

image

**Comment by Vitalis95** [comment]
@rdstern , have a look at it

**Comment by rdstern** [comment]
@Vitalis95 good progress.  
a) seems ok
b) Seems ok
c) Still gives error message, but produces data frame.  
c) Does not produce the key.  If it did, then it could also provide the message about the key in the output window.  Maybe the error message concerns the key?

**Comment by Vitalis95** [comment]
@rdstern , have a look at it

**Review by rdstern — DISMISSED** [review]
@Vitalis95 full marks for persistence.  Thanks.  @lloyddewit over to you.

File: instat/dlgInventoryPlot.vb [hunk 1]
```diff
@@ -306,22 +331,36 @@ Public Class dlgInventoryPlot
...
     End Sub
 
     Private Sub ucrChkDetails_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkDetails.ControlValueChanged
         If ucrChkDetails.Checked Then
+            ucrBase.clsRsyntax.ClearCodes()
             ucrBase.clsRsyntax.AddToAfterCodes(clsClimaticDetails, iPosition:=2)
             clsClimaticDetails.iCallType = 2
         Else
             ucrBase.clsRsyntax.RemoveFromAfterCodes(clsClimaticDetails)
+
```
Comment (lloyddewit) [inline instat/dlgInventoryPlot.vb:363]:
```suggestion
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Vitalis95 Thank you, looks good, just a few very small comments

**Review by rdstern — CHANGES_REQUESTED** [review]
@Vitalis95 looks fine to me, but I have a very small cosmetic change I am afraid.  In the Key, could you make the `From` the first element.  So it goes From, Station, Element, Level.

Thanks

**Comment by Vitalis95** [comment]
@rdstern , its fine

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**